pulsar/internal/batch_builder.go

```diff
@@ -66,7 +66,7 @@ func NewBatchBuilder(maxMessages uint, producerName string, producerID uint64,
 		maxMessages = DefaultMaxMessagesPerBatch
 	}
 	bb := &BatchBuilder{
-		buffer:       NewBuffer(4096),
+		buffer:       NewBuffer(MaxBatchSize),
```

Comment:
That's increasing the min memory size for each producer from 4KB to 1MB. That would be very bad for overall memory usage.